pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -40,8 +40,21 @@ type BuildRunSpec struct {
 	// image would be pushed to. It will overwrite the output image in build spec
 	// +optional
 	Output *Image `json:"output,omitempty"`
+
+	// Used for canceling a buildrun (and maybe more later on).
+	// +optional
+	Status BuildRunSpecStatus `json:"status,omitempty"`
```

Comment:
Hi @gabemontero, when I discussed this today with our UX engineers, the naming of this field caused confusion. `status` is well known to be the (for the k8s user) read-only part of an object. It was not immediately clear that this field was inside the `spec`. As such, would a different name like `targetStatus` or `targetReason` be maybe more appropriate?

Reply:
hey @SaschaSchwarze0 yeah I just copied what existed upstream for tekton and taskrun as a starting point

I'm fine with tweaking it, and sympathize wrt the confusion factor

When I circle back to this (hopefully tomorrow or the day after this week) I'll either go with one of those two or see if I come up with an alternative I at least like better and propose that here

Reply:
OK @SaschaSchwarze0 so back to this today.  I tried to take a step back, and in doing so, revisited the tekton `PipelineRun` version of this field, since they actually have more than one choice there vs. `TaskRun`.

For reference:  https://github.com/tektoncd/pipeline/blob/c8dc797cf5a6f11f90cb742d014470a444fcdc60/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L233-L248

So things I got from that:
- each of the constant's godoc mention the `user` and their intent to change from whatever state .... this got me thinking of things like `override` and `userOverride` ... `override` feels more indicative at least to me of what the user is doing than `target`
- but how to capture the thing being overridden .... `state` ?  `status` ? `reason` ? .... the `reason` field is the final destination of the setting, but that feels like an implementation detail, and not what the user is thinking of when the action is being initiated

So with all that preamble:
- `overrideState`
- `overrideStatus`
- `stateOverride`
- `statusOverride`
- in the end, adding `user` to any of those seemed useful

I think the winner for me, at least for alternatives I can provide, is either `statusOverride` or `overrideStatus` 

I'd say if you and your UX engineers could debate those two against say `targetStatus` (assuming you do not disagree with me that is better than `targetResaon`) and we see if there is a consensus.  I'm fine with `targetStatus` if you all do not like one of my status override ones.

WDYT?

Reply:
Hm, I see you spend a lot time into that. :-)

So, I might be thinking differently tomorrow (this question allows many opinions), but for today, I like the term `state` because it's not `status` as this term is kind of reserved by k8s. And `reason` would become a problem if we ever have more than one entry in our conditions array.

I can also understand `override`.

I could now say let's also ask A and B, but then we probably end up with two more opinions. :-) On the other hand, it's about our API, so its probably needed to ask around.

Reply:
I could live with `overrideState` or `stateOverride`.

So yeah, see how you feel tomorrow, and we'll go from there.